src/osrm/osrm.cpp

```diff
@@ -15,7 +16,18 @@ namespace osrm
 
 // Pimpl idiom
 
-OSRM::OSRM(engine::EngineConfig &config) : engine_(std::make_unique<engine::Engine>(config)) {}
+OSRM::OSRM(engine::EngineConfig &config)
+{
+    if (engine::Engine<engine::algorithm::CoreCH>::CheckCompability(config))
+    {
+        engine_ = std::make_unique<engine::Engine<engine::algorithm::CoreCH>>(config);
+    }
+    else
+    {
+        // Will fail hard if there is no CH data
```

Comment:
Better to check compatibility for `CH` and throw exception if not, otherwise fail can be somewhere deep in routing code